flocker/acceptance/endtoend/test_installer.py

```diff
@@ -0,0 +1,319 @@
+
+PARAMETERS = [
+    {
+        'ParameterKey': 'KeyName',
+        'ParameterValue': os.environ['KEY_NAME']
+    },
+    {
+        'ParameterKey': 'AccessKeyID',
+        'ParameterValue': os.environ['ACCESS_KEY_ID']
+    },
+    {
+        'ParameterKey': 'SecretAccessKey',
+        'ParameterValue': os.environ['SECRET_ACCESS_KEY']
+    }
+]
+
+COMPOSE_NODE0 = '/home/ubuntu/postgres/docker-compose-node0.yml'
+COMPOSE_NODE1 = '/home/ubuntu/postgres/docker-compose-node1.yml'
+RECREATE_STATEMENT = 'drop table if exists test; create table test(i int);'
+INSERT_STATEMENT = 'insert into test values(1);'
+SELECT_STATEMENT = 'select count(*) from test;'
+
+CLOUDFORMATION_STACK_NAME = 'testinstallerstack'
+S3_CLOUDFORMATION_TEMPLATE = (
+    'https://s3.amazonaws.com/'
+    'installer.downloads.clusterhq.com/flocker-cluster.cloudformation.json'
+)
+
+
+def remote_command(node, *args):
+    command_output = []
+    d = run_ssh(
+        reactor,
+        'ubuntu',
+        node,
+        args,
+        handle_stdout=command_output.append
+    )
+    d.addCallback(
+        lambda process_result: (process_result, command_output)
+    )
+    return d
+
+
+def remote_docker_compose(client_ip, docker_host, compose_file_path, *args):
+    docker_compose_output = []
+    d = run_ssh(
+        reactor,
+        'ubuntu',
+        client_ip,
+        ('DOCKER_HOST={}'.format(docker_host),
+         'docker-compose', '--file', compose_file_path) + args,
+        handle_stdout=docker_compose_output.append
+    )
+    d.addCallback(
+        lambda process_result: (process_result, docker_compose_output)
+    )
+    return d
+
+
+def remote_postgres(client_ip, host, command):
+    postgres_output = []
+    d = deferLater(
+        reactor,
+        60,
+        run_ssh,
+        reactor,
+        'ubuntu',
+        client_ip,
+        ('psql', 'postgres://flocker:flocker@' + host + ':5432',
+         '--command={}'.format(command)),
+        handle_stdout=postgres_output.append
+    )
+    d.addCallback(
+        lambda process_result: (process_result, postgres_output)
+    )
+    return d
+
+
+def cleanup(test_case, local_certs_path):
+    certificates_path = FilePath(local_certs_path)
+    cluster_cert = certificates_path.child(b"cluster.crt")
+    user_cert = certificates_path.child(b"user1.crt")
+    user_key = certificates_path.child(b"user1.key")
+    user_credential = UserCredential.from_files(user_cert, user_key)
+    cluster = Cluster(
+        control_node=ControlService(
+            public_address=test_case.control_node_ip.encode("ascii")),
+        nodes=[],
+        treq=treq_with_authentication(
+            reactor, cluster_cert, user_cert, user_key),
+        client=FlockerClient(
+            reactor, test_case.control_node_ip.encode("ascii"),
+            REST_API_PORT, cluster_cert, user_cert, user_key
+        ),
+        certificates_path=certificates_path,
+        cluster_uuid=user_credential.cluster_uuid,
+    )
```

Comment:
:question: from this branch and #2443 it looks like `Cluster` could really use a convenience method/function for constructing.